fenix/app/src/main/java/org/mozilla/fenix/home/ToolbarView.kt

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home
+
+import android.content.Context
+import android.view.Gravity
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.coordinatorlayout.widget.CoordinatorLayout
+import androidx.core.view.updateLayoutParams
+import mozilla.components.support.ktx.android.content.res.resolveAttribute
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.toolbar.ToolbarPosition
+import org.mozilla.fenix.databinding.FragmentHomeBinding
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.home.sessioncontrol.ToolbarInteractor
+import org.mozilla.fenix.utils.ToolbarPopupWindow
+import java.lang.ref.WeakReference
+
+/**
+ * View class for setting up the home screen toolbar.
+ */
+class ToolbarView(
+    private val binding: FragmentHomeBinding,
+    private val context: Context,
+    private val interactor: ToolbarInteractor,
+) {
+    init {
+        updateLayout(binding.root)
+    }
+
+    /**
+     * Setups the home screen toolbar.
+     */
+    fun build() {
```

Comment:
I am pretty sure I can call everything in here in `init`, but need additional confirmation